lib/log_ratelimit.c

```diff
@@ -31,6 +33,8 @@ struct log_ratelimit_state {
 	uint32_t printed;
 	uint32_t suppressed;
 	uint64_t end;
+	uint32_t log_level;
+	char     *block_name;
```

Comment:
Let make `char block_name[16]`as suggested [here](https://github.com/AltraMayor/gatekeeper/issues/497#issuecomment-872262822). This avoids memory allocation/deallocation, which simplifies the code.